File: app/src/main/java/uz/ilhomjon/kirakashgo/taximetr/websocket/MyWebSocketClient.java
package uz.ilhomjon.kirakashgo.taximetr.websocket;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import org.java_websocket.client.WebSocketClient;
import org.java_websocket.handshake.ServerHandshake;

import java.net.URI;
import java.util.ArrayList;

import uz.ilhomjon.kirakashgo.presentation.models.OrdersSocketResponse;
//
//public class MyWebSocketClient extends WebSocketClient {
//    private static final String TAG="MyWebSocketClient";
//    public MyWebSocketClient(URI serverUri) {
//        super(serverUri);
//    }
//
//    @Override
//    public void onOpen(ServerHandshake handshakedata) {
//        // Websocket ochilganda bajariladigan amallar
//        Log.d(TAG, "onOpen: ");
//    }
//
//    @Override
//    public void onMessage(String message) {
//        // Yangi xabar keldi
//        Log.d(TAG, "onMessage: "+message);
//        //
//        ArrayList<OrdersSocketResponse> list = new ArrayList<>();
//
//    }
//
//    @Override
//    public void onClose(int code, String reason, boolean remote) {
//        // Websocket yopilganda bajariladigan amallar
//        Log.d(TAG, "onClose: ");
//    }
//
//    @Override
//    public void onError(Exception ex) {
//        // Xatolik yuz berdi
//        Log.d(TAG, "onError: "+ex.getMessage());
//    }
//}

File: app/src/main/java/uz/ilhomjon/kirakashgo/taximetr/websocket/MyWebSocketClient.kt
package uz.ilhomjon.kirakashgo.taximetr.websocket

import android.util.Log
import androidx.lifecycle.MutableLiveData
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import org.java_websocket.client.WebSocketClient
import org.java_websocket.handshake.ServerHandshake
import uz.ilhomjon.kirakashgo.presentation.models.OrdersSocketResponse
import java.net.URI

class MyWebSocketClient(serverUri: URI) : WebSocketClient(serverUri) {
    private val TAG = "MyWebSocketClient"

    override fun onOpen(handshakedata: ServerHandshake) {
        // Websocket ochilganda bajariladigan amallar
        Log.d(TAG, "onOpen: ")
    }

    companion object{
        val ordersLiveData = MutableLiveData<ArrayList<OrdersSocketResponse>>()
    }
    override fun onMessage(message: String) {
        // Yangi xabar keldi
        Log.d(TAG, "onMessage: $message")
        //
        val list = ArrayList<OrdersSocketResponse>()
        val type = object : TypeToken<ArrayList<OrdersSocketResponse>>(){}.type
        list.addAll(Gson().fromJson(message, type))
        ordersLiveData.postValue(list)
    }

    override fun onClose(code: Int, reason: String, remote: Boolean) {
        // Websocket yopilganda bajariladigan amallar
        Log.d(TAG, "onClose: ")
    }

    override fun onError(ex: Exception) {
        // Xatolik yuz berdi
        Log.d(TAG, "onError: ${ex.message}")
    }
}